fix(OnboardingChecklist): improve height handling and button positioning

jpyzio123 opened the pull request:
Changes: 

- removed hardcoded height for completed checklist container and logic for the opened/closed height. Please let me know if that is needed as I didn't fully understand it and the logic caused a bug where on subsequent renders the opened && completed list height was set to 0/72px since it didn't render in full and saved incorrect value in state.

- changed styles for the completed state of the list when greeting is not provided for the items to align

- changed background colour of the completed list according to new designs


Design: https://www.figma.com/design/LvFTBisF7NAkZQTyyppDbi/Onboarding---Checklist?node-id=824-9809&m=dev 

<!--- Issue number will be inserted automatically if properly defined -->
Resolves: #1626 

## Description

## Storybook

<!--- Branch name will be inserted automatically -->
https://feature-1626--613a8e945a5665003a05113b.chromatic.com

## Checklist

**Obligatory:**

- [ ] Self review (use this as your final check for proposed changes before requesting the review)
- [ ] Add correct label
- [ ] Assign pull request with the correct issue


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Style**
  - Improved visual spacing and alignment in the onboarding checklist, including updated padding and button positioning.
  - Added new styling for completion messages without a greeting.
  - Enhanced toggle button states with distinct positioning for open and closed modes.

- **Refactor**
  - Refined height calculation and management in the onboarding checklist for consistent container resizing and smoother transitions.

No changes to the public interface or exported components.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

marcinsawicki commented:
Something is not good with the align of elements in the complete state:
old:
<img width="1318" alt="Screenshot 2025-04-22 at 08 42 22" src="https://github.com/user-attachments/assets/787830ac-4169-43f5-bdf9-46dd15d05a3e" />

new:
<img width="1318" alt="Screenshot 2025-04-22 at 08 42 29" src="https://github.com/user-attachments/assets/2cfa16de-5846-4ae4-bf12-f85a2fdfd262" />

marcinsawicki reviewed (approved):
The changes look good, only the text alignment needs to be corrected + the opening and closing logic needs to be double checked

jpyzio123 commented:
> 754fb27

@marcinsawicki  good spot, after talking to the designer I changed the paddings and fixed container height for text to align with default values
